cmd/syncthing/main.go

```diff
@@ -924,27 +898,55 @@ func printHashRate() {
 	l.Infof("Single thread hash performance is ~%.*f MB/s", decimals, hashRate)
 }
 
-func loadConfig(cfgFile string) (*config.Wrapper, string, error) {
-	info, err := os.Stat(cfgFile)
+func loadConfig() (*config.Wrapper, error) {
+	cfgFile := locations[locConfigFile]
+	cfg, err := config.Load(cfgFile, myID)
+
 	if err != nil {
-		return nil, "", err
+		l.Infoln("Error loading config file; using defaults for now")
+		myName, _ := os.Hostname()
+		newCfg := defaultConfig(myName)
+		cfg = config.Wrap(cfgFile, newCfg)
 	}
-	if !info.Mode().IsRegular() {
-		return nil, "", errors.New("configuration is not a file")
```

Comment:
That check is now unnecessary, since `loadConfig()` always returns a config object and all errors, and is directly used by functions who don't care about the file actually existing. The ones that do care will use `loadOrCreateConfig()` which checks for all errors (lines 917-922).

I think that covers every case, right?